scripts/experiments_tabular/maf.jl

```diff
@@ -0,0 +1,92 @@
+using DrWatson
+@quickactivate
+using ArgParse
+using GenerativeAD
+using StatsBase: fit!, predict, sample
+using BSON
+
+s = ArgParseSettings()
+@add_arg_table! s begin
+   "max_seed"
+        required = true
+        arg_type = Int
+        help = "seed"
+    "dataset"
+        required = true
+        arg_type = String
+        help = "dataset"
+end
+parsed_args = parse_args(ARGS, s)
+@unpack dataset, max_seed = parsed_args
+
+modelname = "MAF"
+function sample_params()
+	par_vec = ([2, 5, 10], 2 .^(4:10), 2:3, ["natural", "random"], [1f-4], [100], [30], [1f-6])
+	argnames = (:nflows, :hdim, :nlayers, :ordering, :lr, :batchsize, :patience, :wreg)
+
+	return (;zip(argnames, map(x->sample(x, 1)[1], par_vec))...)
+end
+
+function fit(data, parameters)
+	idim = size(data[1][1], 1)
+
+	model = GenerativeAD.Models.MAF(
+				parameters.nflows,
+				idim,
+				parameters.hdim,
+				parameters.nlayers,
+				parameters.ordering)
+
+	try
+		global info, fit_t, _, _, _ = @timed fit!(model, data, parameters)
+ 	catch e
+		@info "Failed training due to \n$e"
+		return (fit_t = NaN,), []
```

Comment:
Consider setting the remaining fields as well, e.g. `model = nothing`, see my comment on model saving.